packages/ethereum/contracts/test/Crypto.t.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.6.0 <0.9.0;
+
+import 'forge-std/Test.sol';
+import './utils/Accounts.sol';
+import 'solcrypto/SECP2561k.sol';
+
+import '../src/Crypto.sol';
+
+contract CryptoProxy is HoprCrypto {
+}
+
+contract Crypto is Test, AccountsFixtureTest, HoprCrypto {
+  struct CurvePoint {
+    uint256 x;
+    uint256 y;
+  }
+
+  SECP2561k secp256k1;
+  CryptoProxy crypto;
+  function setUp() public {
+    // use dummy proxy to per-method have gas measurements
+    crypto = new CryptoProxy();
+    secp256k1 = new SECP2561k();
+  }
+
+  function testPointToAddress() public {
+    (bool success, bytes memory returnValue) = address(this).staticcall(abi.encodeWithSelector(HoprCrypto.pointToAddress.selector, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+
+    assertTrue(success);
+    assertEq(address(uint160(uint256(bytes32(returnValue)))), accountA.accountAddr);
+  }
+
+  function testIsCurvePoint() public { 
+    assertTrue(crypto.isCurvePointInternal(0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+  }
+
+  function testRevert_NoCurvePoint() public {
+    assertFalse(crypto.isCurvePointInternal(0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75));
+  }
+
+  function testScalarTimeBasepoint() public {
+    assertEq(crypto.scalarTimesBasepoint(accountA.privateKey), accountA.accountAddr);
+  }
+
+  function testEcAddPointAddition() public {
+    CurvePoint memory p = CurvePoint(0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75,0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5);
+    CurvePoint memory q = CurvePoint(0xba5734d8f7091719471e7f7ed6b9df170dc70cc661ca05e688601ad984f068b0,0xd67351e5f06073092499336ab0839ef8a521afd334e53807205fa2f08eec74f4);
+    CurvePoint memory r = CurvePoint(0x9d9031e97dd78ff8c15aa86939de9b1e791066a0224e331bc962a2099a7b1f04,0x64b8bbafe1535f2301c72c2cb3535b172da30b02686ab0393d348614f157fbdb);
+
+    CurvePoint memory p_q = CurvePoint(0x551c7c46a964dec7edd8a5cedc557ebce43cc3f70ff481bdcfbd4e86d435c2ba,0x16883c2c7e2527800aa21a8420f8af48eafb2594d00e0f7e9e7d11a938b9a168);
+    CurvePoint memory q_r = CurvePoint(0x85744a09c2839969dd8aa41b3577e9ffa28bee884165b880ca4050c0c3a0083e,0xbf55c694c111642da3ac0017e44ed12b93798eb9cbd3b14b4db1227c85e58f6d);
+    
+    (uint256 maybe_p_q_x, uint256 maybe_p_q_y) = crypto.ecAdd(p.x, p.y, q.x, q.y, 0);
+    assertEq(p_q.x, maybe_p_q_x);
+    assertEq(p_q.y, maybe_p_q_y);
+
+    (uint256 maybe_q_r_x, uint256 maybe_q_r_y) = crypto.ecAdd(q.x, q.y, r.x, r.y, 0);
+    assertEq(q_r.x, maybe_q_r_x);
+    assertEq(q_r.y, maybe_q_r_y);
+  }
+
+  function testEcAddPointDouble() public {
```

Comment:
Same here, let's tests also the failure modes.

Reply:
changed the method description accordingly